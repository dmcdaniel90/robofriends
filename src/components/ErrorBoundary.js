import React, { useState, useEffect } from "react";

function ErrorBoundary (props) {

  const [hasError, setHasError] = useState(false);

  useEffect(() => {
    if (hasError) {
      setHasError(true);
    }
  }, [hasError])

  return(
    hasError ? <h1>Oops. That is not good</h1> : props.children
  )
}

export default ErrorBoundary;