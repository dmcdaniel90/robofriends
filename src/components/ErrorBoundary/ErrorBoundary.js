function ErrorBoundary (props) {

  let { error } = props;

  return(
    error ? <h1>Oops. That is not good</h1> : props.children
  )
}

export default ErrorBoundary;