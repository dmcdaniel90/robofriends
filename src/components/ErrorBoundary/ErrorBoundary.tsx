function ErrorBoundary (props: { error: Error, children: React.ReactNode }): React.ReactNode {

  let { error } = props;

  return(
    error ? <h1>Oops. That is not good</h1> : props.children
  )
}

export default ErrorBoundary;