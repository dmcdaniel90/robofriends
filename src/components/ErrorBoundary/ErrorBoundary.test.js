import {render} from '@testing-library/react'
import ErrorBoundary from './ErrorBoundary';

describe('ErrorBoundary', () => {
  it('renders the ErrorBoundary component', () => {
    render(<ErrorBoundary />)
  })

  it('renders the ErrorBoundary component with an error', () => {
    const error = new Error('test error');
    const { container } = render(<ErrorBoundary error={error} />);
    expect(container).toMatchSnapshot();
  })
})