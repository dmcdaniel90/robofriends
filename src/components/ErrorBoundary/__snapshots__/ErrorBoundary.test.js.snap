// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary renders the ErrorBoundary component with an error 1`] = `
<div>
  <h1>
    Oops. That is not good
  </h1>
</div>
`;
